release_draft.md

```diff
@@ -0,0 +1,305 @@
+export DEBEMAIL="<username>@openrobotics.org"
+export DEBFULLNAME="<Your full name>"
+```
+**Note:** the two above exported variables can be added to `.bashrc` to have
+them configured automatically on every run.
+
+
+#### Credentials
+
+There is a small number of credentials needed to interact with the release
+process:
+
+ * S3 access to open robotics:
+  ```
+  s3cmd --configure
+  ```
+  If you don't have AWS credentials please contact @j-rivero or @nuclearsandwich who will help set you up.
+
+ * Release token: magic sequence of characters needed while running `release.py`
+   to interact with `build.osrfoundation.org`:
+   ```
+   Check your company password manager for 'build.osrfoundation.org token'
+   ```
+
+### Preparing Ignition Code
+
+## Update code version
+
+First step to get a new release ready is to update the current code (upstream)
+version to a new one (more information in the [versioning](#versioning)). This
+bump could be in the major number (non compatible changes), minor number (new
+features), patch number (patches and bugfixes).
+
+1. **Bumping major number** of the version implies some work to have the
+[metadata](#metadata-for-releasing) updated correctly. There is a [dedicated
+document](releasing/bump_major.md) to do this before going further in this
+document.
+
+   1. To update the upstream version a local checkout of the Ignition library is
+   needed. A new branch is required to submit changes:
+
+  ```bash
+  # version X.Y.Z
+  git checkout -b bump_to_X_Y_Z
+  ```
+
+  2. The current upstream version can be found in `CMakeLists.txt` file
+  following the CMake `project declaration`:
+
+  ```cmake
+  # Ignition library named foo and version X.Y.Z
+  project(ignition-fooX VERSION X.Y.Z)
+  ```
+
+  Stable releases can modify the X, Y or Z directly while prereleases will need to
+  include the preX (X number starts with 1) suffix in the `ign_configure_project`:
+  ```cmake
+  # first prerelease of a serie, number 1
+  ign_configure_project(VERSION_SUFFIX pre1)
+  ```
+
+  3. Together with bumping the version number **updating the Changelog and Migration
+  documents** is highly recommended. `Changelog.md` file and `Migration.md` files
+  are located at the top level of every Ignition library. Modify them as needed.
+
+  ```bash
+  git commit CMakeList.xt
+  git commit Changelog.md
+  git commit Migration.md
+  git push orign ...
+  ```
+
+  4. Open a pull request for reviewing ([example PR](https://github.com/ignitionrobotics/ign-physics/pull/132)).
+  Including a link is recommended comparing the current release branch to the
+  latest release. Releases are tagged in GitHub repositories with the scheme
+  `ignition-fooX_X.Y.Z` where foo is the name of the Ignition library and X.Y.Z
+  the code version. ([example of a branch comparison](https://github.com/ignitionrobotics/ign-gazebo/compare/ignition-gazebo3_3.5.0...ign-gazebo3)
+
+  5. Check that the CI job named `*-abichecker-*` is fine since its specially
```

Comment:
Does the ABI checker check against the latest released tag or against the target release branch? I believe it's still the latter, right? See https://github.com/ignition-tooling/release-tools/issues/24#issuecomment-611519063.

If so, the ABI checker on the release PR will never fail, because that PR doesn't change C++ code...

Reply:
you are right, removed d8b482c4d83a394d0c82a1b654e9c0f2c60cae92